### DISCUSSION TITLE
Customising using settings.yml

### DISCUSSION THREAD

**Discussion by samgaw58** [Stirling-Tools/Stirling-PDF#393] [General]
I have followed the simple instructions on customising the setup, both on the install version and the latest docker image. It made not one jot of difference. Is there another file I need to edit that I'm missing?

**Comment by Frooodle** [comment]
Can you show what your settings file contains?
Or how you are running docker imag etc?

**Comment by samgaw58** [comment]
Here is a copy of the settings.yml file. I have the same settings in both the installed version & the docker image (version 0.14.3)

# Welcome to settings file
# Remove comment marker # if on start of line to enable the configuration
# If you want to override with environment parameter follow parameter naming SECURITY_INITIALLOGIN_USERNAME

security:
  enableLogin: false # set to 'true' to enable login
  csrfDisabled: true

system:
  defaultLocale: 'en-GB' # Set the default language (e.g. 'de-DE', 'fr-FR', etc)
  googlevisibility: false # 'true' to allow Google visibility (via robots.txt), 'false' to disallow

  appName: My PDF # Application's visible name
  appNameNavbar: My PDF # Name displayed on the navigation bar

endpoints:
  toRemove: [] # List endpoints to disable (e.g. ['img-to-pdf', 'remove-pages'])
  groupsToRemove: [] # List groups to disable (e.g. ['LibreOffice'])

metrics:
  enabled: true # 'true' to enable Info APIs (`/api/*`) endpoints, 'false' to disable
AutomaticallyGenerated:

**Comment by samgaw58** [comment]
The docker version is running on a Ubuntu 22.04 LXC container & the installed version is running on a 22.04 VM.

**Comment by Frooodle** [comment]
Ahh i think its a error in my config generation
the UI section should look like
```
ui:
  appName: exampleAppName # Application's visible name
  homeDescription: I am a description # Short description or tagline shown on homepage.
  appNameNavbar: navbarName # Name displayed on the navigation bar
  ```
  
  
I dont see that ui: in your above

**Reply by samgaw58** [reply]
As soon as I added ui: into the yaml file everything changed to how I wanted it. Many many thanks for your help.

**Comment by Frooodle** [comment]
also im assuming your is correctly indented and just copied badly can you confirm?

**Comment by kalipso-cyber** [comment]
Hey,

sorry, to dig this up, but I'm also having trouble doing any sort of customization. Reading this thread did not really help me, so I'm hoping for some clarification. In the GitHub Readme, it says:

> Stirling PDF allows easy customization of the app. Includes things like
> 
>     Custom application name
>     Custom slogans, icons, HTML, images CSS etc (via file overrides)
> 
> There are two options for this, either **using the generated settings file settings.yml** This file is located in the /configs directory and follows standard YAML formatting

Where exactly is this `/configs` folder supposed to be found? I built the app with docker-compose, and I didn't find a `/configs` folder anywhere. I also deployed it using a Dockerfile, but there still is no `/configs` directory to be found. Similarly, I couldn't find a `settings.yml` being generated anywhere. Did I misunderstand the instructions?

Trying to pass the environment variables directly with the `docker run` command throws an error:

```
$ sudo docker run -d \
  -p 8080:8080 \
  -v /location/of/trainingData:/usr/share/tessdata \
  -v /location/of/extraConfigs:/configs \
  -v /location/of/logs:/logs \
  -e DOCKER_ENABLE_SECURITY=false \
  -e INSTALL_BOOK_AND_ADVANCED_HTML_OPS=false \
  -e LANGS=en_GB,de_DE,en_US \
  -e UI_APP_NAME=Stirling PDF \
  -e UI_HOME_DESCRIPTION=Custom description \
  -e UI_APP_NAVBAR_NAME=Custom name \
  --name stirling-pdf \
  frooodle/s-pdf:latest
docker: invalid reference format: repository name (library/PDF) must be lowercase.
See 'docker run --help'.
```

(The same command works flawlessly when I omit the last three environment variables.)

So in short, I tried the steps listed at the bottom of https://stirlingtools.com/docs/Advanced%20Configuration/How%20to%20add%20configurations/, but couldn't get either to work. Could somebody point me in the right direction?

Also, I think there's some confusion under this link: https://stirlingtools.com/docs/Advanced%20Configuration/System%20and%20Security/. I'm pretty sure that three out of four of the parameters listed on the top of the page hold functionality that's different from what is mentioned there ;)

**Reply by Frooodle** [reply]
Please follow GitHub not Stirling tools 

Also in your docker compose you showed those default paths like location/of/files etc
Have you changed these to match any location on your system?

**Reply by kalipso-cyber** [reply]
Ok.

No I haven't changed those file paths. I simply followed the instructions under "[How to use](https://github.com/Stirling-Tools/Stirling-PDF#how-to-use)" assuming they'd tell me what to do. Should I actually just use one of the [exampleYmlFiles](https://github.com/Stirling-Tools/Stirling-PDF/tree/main/exampleYmlFiles) instead? If so, what's the difference between the `fat` and regular versions?

**Reply by Frooodle** [reply]
Change /location/of/extraConfigs etc to
./extraConfigs 


That location path is basically a big CHANGE ME 
For the location of a folder on your host machine to store those files

**Reply by Frooodle** [reply]
https://github.com/Stirling-Tools/Stirling-PDF/commit/3576c32c5273c1a91bb731f4591306274e9e525c

updated docs

**Reply by kalipso-cyber** [reply]
Makes sense.. So the files are generated in that location, and once I first started the app, I subsequently add the changes in the now-generated `settings.yml`, and then start the app again?
